src/main/java/io/cdap/plugin/gcp/gcs/StorageClient.java

```diff
@@ -44,6 +44,7 @@
 public class StorageClient {
   private static final Logger LOG = LoggerFactory.getLogger(StorageClient.class);
   private final Storage storage;
+  private List<GCSPath> undoBucket = new ArrayList<>();
```

Comment:
This makes the StorageClient stateful, which is bad. It assumes somebody will call methods in a certain order and behaves incorrectly otherwise.

Just make createBucket return the Bucket it created, or null if it didn't do anything